packages/remix-server-runtime/server.ts

```diff
@@ -265,7 +302,214 @@ async function handleDataRequestRR(
+      { data: await unwrapResponse(response), status: response.status },
+      response.headers,
+    ];
+  } catch (err) {
+    handleError(err);
+    let error = isResponse(err)
+      ? new ErrorResponseImpl(
+          err.status,
+          err.statusText,
+          await unwrapResponse(err)
+        )
+      : err;
+    return [{ error }, new Headers()];
+  }
+}
+
+async function singleFetchLoaders(
+  handlerUrl: URL,
+  routesToLoad: string | null,
+  staticHandler: StaticHandler,
+  matches: RouteMatch<ServerRoute>[] | null,
+  loadContext: AppLoadContext,
+  handleError: (err: unknown) => void,
+  serverMode: ServerMode,
+  build: ServerBuild
+): Promise<[SingleFetchResults, Headers]> {
+  let context: StaticHandlerContext;
+  try {
+    let handlerRequest = new Request(handlerUrl);
+    let loadRouteIds = routesToLoad ? routesToLoad.split(",") : undefined;
+
+    let result = await staticHandler.query(handlerRequest, {
+      requestContext: loadContext,
+      loadRouteIds,
+    });
+    if (isResponse(result)) {
+      // We don't really know which loader this came from, so just stick it at
+      // a known match
+      let routeId =
+        matches?.find((m) =>
+          routesToLoad
+            ? routesToLoad.split(",").includes(m.route.id)
+            : m.route.module.loader
+        )?.route.id || "root";
+      return [
+        {
+          [routeId]: {
+            redirect: result.headers.get("Location")!,
+            status: result.status,
+            revalidate: result.headers.has("X-Remix-Revalidate"),
+            reload: result.headers.has("X-Remix-Reload-Document"),
+          },
+        },
+        result.headers,
+      ];
+    }
+    context = result;
+  } catch (error: unknown) {
+    handleError(error);
+    return [{ root: { error } }, new Headers()];
+  }
+
+  // Sanitize errors outside of development environments
+  if (context.errors) {
+    Object.values(context.errors).forEach((err) => {
+      // @ts-expect-error This is "private" from users but intended for internal use
+      if (!isRouteErrorResponse(err) || err.error) {
+        handleError(err);
+      }
+    });
+    context.errors = sanitizeErrors(context.errors, serverMode);
+
+    // TODO: Feels hacky - we need to un-bubble errors here since they'll be
+    // bubbled client side.  Probably better to throw a flag on query() to not
+    // do this in the first place
+    let mostRecentError: [string, unknown] | null = null;
+    for (let match of context.matches) {
+      let routeId = match.route.id;
+      if (context.errors[routeId] !== undefined) {
+        mostRecentError = [routeId, context.errors[routeId]];
+      }
+      if (
+        build.assets.routes[routeId]?.hasLoader &&
+        context.loaderData[routeId] === undefined &&
+        mostRecentError
+      ) {
+        context.errors[mostRecentError[0]] = undefined;
+        context.errors[routeId] = mostRecentError[1];
+        mostRecentError = null;
+      }
+    }
```

Comment:
Don't love this - probably should move this behind a flag in `query` to turn off error bubbling.  The other option is to run a bunch of `queryRoutes` in parallel but I kinda like this mimicing document requests at the moment 🤷